core/domain/domain.go

```diff
@@ -0,0 +1,24 @@
+package domain
+
+import (
+	"context"
+	"time"
+)
+
+type Repository interface {
+	Create(ctx context.Context, domain Domain) error
+	Get(ctx context.Context, id string) (Domain, error)
+	Update(ctx context.Context, domain Domain) (Domain, error)
+	List(ctx context.Context, flt Filter) ([]Domain, error)
+	Delete(ctx context.Context, id string) error
+}
+
+type Domain struct {
+	ID         string
+	Name       string
+	OrgID      string
+	Token      string
+	Verified   bool
+	VerifiedAt time.Time
+	CreatedAt  time.Time
```

Comment:
How about having an expires at column as well?

Reply:
was thinking of keeping it abstract since all the verified domains will not require that field, the cron job can be just used to remove the pending state domains with created-at time being more than 7 days